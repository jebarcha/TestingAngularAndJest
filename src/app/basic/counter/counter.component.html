<h1>Counter <small>{{ counter }}</small></h1>
<hr>

<button class="btn btn-primary mx-1" (click)="increaseBy(+1)">+1</button>
<button class="btn btn-primary mx-1" (click)="increaseBy(-1)">-1</button>