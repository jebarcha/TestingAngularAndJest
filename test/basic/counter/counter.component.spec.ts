import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from '../../../src/app/basic/counter/counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ CounterComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should component to match snapshot', () => { 
    expect( compiled ).toMatchSnapshot();
  });

  test('should counter increment based on the argument', () => { 
    
    component.increaseBy(5);
    expect(component.counter).toBe(15);
  });

  test('should increment and decrement by 1 when clicking the buttons', () => { 
    
    const buttons = compiled.querySelectorAll('button');
    buttons[0].click();
    buttons[0].click();
    expect(component.counter).toBe(12);

    buttons[1].click();
    expect(component.counter).toBe(11);

  });

  test('should update the label when the counter change', () => { 
    
    component.increaseBy(10);
    fixture.detectChanges();

    const h1 = compiled.querySelector('h1');
    expect(h1?.textContent).toContain('20');

  });

});
