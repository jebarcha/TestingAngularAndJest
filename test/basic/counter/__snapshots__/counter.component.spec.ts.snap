// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent should component to match snapshot 1`] = `
<div
  id="root1"
>
  <h1>
    Counter 
    <small>
      10
    </small>
  </h1>
  <hr />
  <button
    class="btn btn-primary mx-1"
  >
    +1
  </button>
  <button
    class="btn btn-primary mx-1"
  >
    -1
  </button>
</div>
`;
